packages/functions/src/service.ts

```diff
@@ -122,7 +122,7 @@ export class FunctionsService implements _FirebaseService {
     // Resolve the region or custom domain overload by attempting to parse it.
     try {
       const url = new URL(regionOrCustomDomain);
-      this.customDomain = url.origin;
+      this.customDomain = url.origin + url.pathname;
```

Comment:
In the case that the pathname is empty, the custom domain will have a trailing `/`. This will result in `_url('foo')` returning a URL looking like `https://mydomain.com//foo`. To fix this, I think we can only append the `pathname` when it's not empty.
```suggestion
      this.customDomain =
        url.origin + (url.pathname === '/' ? '' : url.pathname);
```